Subject: Linux-RT 2.6.24-rc2-rt1
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From sebastien.dugue@bull.net Thu Oct 11 11:32:58 2007
Date: Thu, 11 Oct 2007 14:24:17 +0200
From: "[UTF-8] Sébastien Dugué" <sebastien.dugue@bull.net>
To: Ingo Molnar <mingo@elte.hu>, Thomas Gleixner <tglx@linutronix.de>,
     Steven Rostedt <rostedt@goodmis.org>
Cc: Linux RT Users <linux-rt-users@vger.kernel.org>,
     linux-kernel <linux-kernel@vger.kernel.org>
Subject: [PATCH] RT: fix spin_trylock_irq

    [ The following text is in the "UTF-8" character set. ]
    [ Your display is set for the "iso-8859-1" character set.  ]
    [ Some characters may be displayed incorrectly. ]


  This patch fixes a bug in spin_trylock_irq() where __spin_trylock_irq() is
picked for regular (non-raw) spinlocks instead of _spin_trylock_irq().

  This results in systematic boot hangs and may have been going unnoticed
for quite some time as it only manifests (aside from a compile warning) when
booting with a NUMA config or when using the Chelsio T3 (cxgb3) driver as
these seems to be the sole users.


Signed-off-by: Sébastien Dugué <sebastien.dugue@bull.net>

---
 include/linux/spinlock.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.24-rc2-rt1/include/linux/spinlock.h
===================================================================
--- linux-2.6.24-rc2-rt1.orig/include/linux/spinlock.h
+++ linux-2.6.24-rc2-rt1/include/linux/spinlock.h
@@ -501,7 +501,7 @@ do {									\
 
 #define spin_trylock_irq(lock)	\
 	__cond_lock(lock, PICK_SPIN_OP_RET(__spin_trylock_irq,		\
-		__spin_trylock_irq, lock))
+		_spin_trylock_irq, lock))
 
 #define spin_trylock_irqsave(lock, flags) \
 	__cond_lock(lock, PICK_SPIN_OP_RET(__spin_trylock_irqsave, 	\
